### DISCUSSION TITLE
Allow rebinding of hotkeys in the editor

### DISCUSSION THREAD

**Discussion by Lyti4n** [ppy/osu#29288] [Ideas]
### Type

Game behaviour

### Bug description

Currently you can't rebind the any of the keys you would actually use frequently such as the circle and slider tool. Considering the fact that I'm left handed and so I use the right side of the keyboard when using the editor, not being able to change these keybinds makes the editor practically unusable for me.

### Version

2024.731.0

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16485392/compressed-logs.zip)

**Comment by peppy** [comment]
Can you not move your hand to the left side of the keyboard?

We may add more binding support in the future but this is the first request we've had for this after almost two decades.

I get you're trying to get yourself the change made, but use of "practically unusable" kind of language is not going to help when there are obvious workarounds you can apply locally.

**Reply by Lyti4n** [reply]
sorry about the language I was just a bit tilted. My setup doesn't have enough room to move my keyboard any further to the right and keeping my right hand on the left side of the keyboard is very uncomfortable for me, the larger issue for me though is the fact that on stable I have a different layout of the new combo, circle tool and slider tool than the default which would require me to relearn the hotkeys which i've learned over a long time, basically it would be extremely annoying for me to relearn that muscle memory over what to me seems like a pretty simple change. Sorry again that I've come off as aggresive during this.

**Reply by peppy** [reply]
The layout should actually be the same, can you better explain what you're finding different?

**Reply by Lyti4n** [reply]
sorry I meant as in the layout that I rebinded my keys to in stable can't be replicated in lazer due to not being able to rebind the circle, slider and new combo tool.

**Reply by peppy** [reply]
Right, didn't realise you could rebind these in stable. We'll aim to replicate this.

![Jump Desktop 2024-08-06 at 04 53 36](https://github.com/user-attachments/assets/fd06d1f3-b2a9-4df0-a362-31446ebac7e9)

**Reply by Lyti4n** [reply]
is there any progress on this? rebinding a handful of things that stable allows you to do seems like a pretty minor change, and its really very annoying for people who are used to custom keybinds like myself.

**Reply by peppy** [reply]
Which did you want to rebind?

**Reply by Lyti4n** [reply]
the new combo toggle and switching between slider and circle tools are my most used, but it probably makes sense to do all of these
<img width="291" height="720" alt="ljdr73gcbcng1" src="https://github.com/user-attachments/assets/4a41a484-c993-403a-bcca-ed86960cdf81" />

**Reply by peppy** [reply]
Sure. What do you have things bound to out of curiosity?

**Reply by Lyti4n** [reply]
im left handed so i have new combo set to "N", select tool set to "L", circle set to "K", and slider set to "J"

**Comment by Lyti4n** [comment]
Sorry to bump this again but it's driving me crazy at this point now that I've fully switched to mapping on Lazer.

**Reply by peppy** [reply]
Yes well this is a pain in the ass to add because of rulesets having different controls displayed.
